app/js/lib/foxbox/api.js

```diff
@@ -148,4 +222,78 @@ export default class API {
 
     return new Promise((resolve) => settings.once('session', () => resolve()));
   }
+
+  /**
+   * Fetches values for the set of getters.
+   *
+   * @return {Promise}
+   * @private
+   */
+  [p.fetchGetterValues]() {
+    // It may happen that all watchers have been unregistered in the meantime,
+    // so let's return early in this case.
+    if (this[p.watchGetters].size === 0) {
+      return Promise.resolve();
+    }
+
+    const selectors = Array.from(this[p.watchGetters].values()).map(
+      ({ id }) => ({ id })
+    );
+
+    return this.put('channels/get', selectors)
+      .then((response) => {
+        Object.keys(response).forEach((key) => {
+          const getter = this[p.watchGetters].get(key);
+          if (!getter) {
+            return;
+          }
+
+          this[p.updateGetterValue](getter, response[key]);
+        });
+      });
+  }
+
+  /**
+   * Updates getter value if needed. If value has changed, appropriate event is
+   * fired.
+   *
+   * @param {{ id: string, value: Object }} getter Getter to update value for.
+   * @param {Object} getterValue Getter value returned from the server.
+   *
+   * @private
+   */
+  [p.updateGetterValue](getter, getterValue) {
+    let valueChanged = false;
+
+    if (!getterValue || !getter.value) {
+      valueChanged = getterValue !== getter.value;
+    } else {
+      const [valueKind] = Object.keys(getterValue);
+      if (valueKind === 'Error') {
+        console.error(
+          'Failed to retrieve value for getter (%s): %o',
+          getter.id,
+          getterValue[valueKind]
+        );
+
+        return;
+      }
+
+      const newValue = getterValue[valueKind];
+      const oldValue = getter.value[valueKind];
+
+      if (newValue && oldValue && typeof newValue === 'object') {
+        // If value is a non-null object, we use their JSON representation to
+        // compare values.
+        valueChanged = JSON.stringify(newValue) !== JSON.stringify(oldValue);
```

Comment:
I'm sure there are more efficient method to do deep object comparison. Also if the properties order is not the same, the comparison will fail. As per the es spec, the objects property order shouldn't be relied on.